handler/src/main/java/io/netty/handler/ssl/SniHandler.java

```diff
@@ -56,7 +58,7 @@
             InternalLoggerFactory.getInstance(SniHandler.class);
     private static final Selection EMPTY_SELECTION = new Selection(null, null);
 
-    protected final AsyncMapping<String, SslContext> mapping;
+    protected final SslContextMapping mapping;
```

Comment:
ATTENTION: I think this is the only thing that breaks the API (unfortunately). If that's OK then I'd actually recommend to make this field private.